corehq/apps/app_manager/views/download.py

```diff
@@ -206,9 +208,13 @@ def download_file(request, domain, app_id, path):
         content_type = None
     response = HttpResponse(content_type=content_type)
 
+    build_profile = request.GET.get('profile')
+    build_profile_access = domain_has_privilege(domain, privileges.BUILD_PROFILES)
     if path in ('CommCare.jad', 'CommCare.jar'):
         set_file_download(response, path)
         full_path = path
+    elif build_profile and build_profile in request.app.build_profiles and build_profile_access:
```

Comment:
This is the desired behavior as per the spec. That way if a profile is deleted, the users will still be able to update the application without reinstalling from scratch.